sound/soc/intel/boards/sof_ssp_amp.c

```diff
@@ -249,9 +251,13 @@ static struct snd_soc_dai_link *sof_card_dai_links_create(struct device *dev,
 	}
 
 	/* codec SSP */
-	links[id].name = devm_kasprintf(dev, GFP_KERNEL, "SSP%d-Codec", ssp_codec);
-	if (!links[id].name)
-		return NULL;
+	if (sof_ssp_amp_quirk & SOF_SSP_VIRTUAL_SSP_DAI)
+		links[id].name = "SSP-SPK";
+	else {
+		links[id].name = devm_kasprintf(dev, GFP_KERNEL, "SSP%d-Codec", ssp_codec);
```

Comment:
@brentlu this doesn't look backwards-compatible with existing topologies. How would we bind the definitions in the existing topologies with a dailink has a different name depending on the kernel version.

I don't see how we can move forward with this.

Reply:
When google uprev the kernel for Chromebooks, topology and firmware will got upgraded, too. There are branches for Chrome production firmware so it's manageable.

Reply:
sorry, no, that's not an acceptable argument @brentlu. Some users put the device in developer mode and install whatever distribution they want. We cannot accept such a generic change because of how Chrome deal with their releases.

Reply:
Also @cujomalainey and others would like to test upstream kernels on Chromebooks *before* the uprev takes place. We cannot assume the installation of newer firmware/topology when testing new kernels, it's not manageable from a kernel CI perspective.

Reply:
@plbossart Understood. Do you think it's acceptable for MTL and later platforms? We could prevent creating DMI quirks simply because different SSP port usage.

Reply:
Oh for MTL the runway is really short @brentlu. We have no released any topologies, but we should indeed consider simplifications if that helps. @ranj063 I guess we'd need your opinion here on changing directions wrt. BE dailink naming.

Reply:
I believe that is the right direction to unify the topology. Once the BE link name is consistent, we can try to use consistent BE id for each interface. Eventually we can have a general topology and each topology is a subset of the general topology.